### DISCUSSION TITLE
Is there a way to ensure a guest using smartcard passthrough and SPICE gets access to the smartcard?

### DISCUSSION THREAD

**Discussion by jg110** [virt-manager/virt-manager#637] [Q&A] [answered]
I've got a Windows 10 guest on a RHEL 9.3 host. I've compiled QEMU with smartcard support and have added a smartcard device in passthrough mode, installed OpenSC in Windows, and confirmed that my smart card is working on RHEL.
Windows recognizes a card reader, but no card. However, if I start the VM in virt-manager and then connect to the running VM using `remote-viewer --spice-smartcard spice://127.0.0.1:5900`, the smartcard becomes visible to Windows. If I close the remote-viewer window, the smartcard disappears. (Edit: I should add that this is with the latest release of virt-manager and virt-viewer on the website, both compiled from source, so versions 4.1.0 and 11.0)

Is there a configuration option to run the viewer part of virt-manager with --spice-smartcard? I'm assuming that the viewer part is basically identical to virt-viewer/remote-viewer. And shouldn't this be done automatically if a VM has a smartcard device?

**Comment by phrdina** [comment, marked as answer]
Quick look at virt-viewer and virt-manager code suggest that virt-manager doesn't support smartcard using spice. There is no code to handle the smartcard spice channel.

So your option is to keep using virt-viewer/remote-viewer or you can try assigning the whole smartcard device from host to VM.
